src/modules/job-exec/job-exec.c

```diff
@@ -845,15 +846,111 @@ static flux_future_t *ns_create_and_link (flux_t *h,
     flux_future_t *f = NULL;
     flux_future_t *f2 = NULL;
 
-    if (!(f = flux_kvs_namespace_create (h, job->ns, job->userid, flags))
-        || !(f2 = flux_future_and_then (f, namespace_link, job))) {
+    if (job->reattach && job->rootref)
+        f = flux_kvs_namespace_create_with (h,
+                                            job->ns,
+                                            job->rootref,
+                                            job->userid,
+                                            flags);
+    else
+        f = flux_kvs_namespace_create (h, job->ns, job->userid, flags);
+
+    if (!f || !(f2 = flux_future_and_then (f, namespace_link, job))) {
         flux_log_error (h, "namespace_move: flux_future_and_then");
         flux_future_destroy (f);
         return NULL;
     }
     return f2;
 }
 
+static void get_rootref_cb (flux_future_t *fprev, void *arg)
+{
+    int saved_errno;
```

Comment:
Should this code for restoring namespaces be in the same source file as the code to save them?

Reply:
i assume you meant "removing namespaces".  Good point, perhaps it should be in `job-exec.c` b/c that's where we create/setup namespaces too.  I'll move it in there.